src/app/wallets/reimburse-failed-usd.ts

```diff
@@ -0,0 +1,80 @@
+import { toSats } from "@domain/bitcoin"
+import {
+  CouldNotFindWalletForWalletCurrencyError,
+  InvalidAccountForWalletIdError,
+} from "@domain/errors"
+import { DisplayCurrency, toCents } from "@domain/fiat"
+import { LedgerTransactionType } from "@domain/ledger"
+import { AmountCalculator, WalletCurrency } from "@domain/shared"
+
+import { WalletsRepository } from "@services/mongoose"
+import * as LedgerFacade from "@services/ledger/facade"
+
+const calc = AmountCalculator()
+
+export const reimburseFailedUsdPayment = async <
+  S extends WalletCurrency,
+  R extends WalletCurrency,
+>({
+  paymentFlow,
+  journalId,
+  accountId,
+}: {
+  paymentFlow: PaymentFlow<S, R>
+  journalId: LedgerJournalId
+  accountId: AccountId
+}): Promise<true | ApplicationError> => {
+  const paymentHash = paymentFlow.paymentHashForFlow()
+  if (paymentHash instanceof Error) return paymentHash
+
+  const metadata: FailedPaymentLedgerMetadata = {
+    hash: paymentHash,
+    type: LedgerTransactionType.Payment,
```

Comment:
shouldn't we have a more precise type for this?

Reply:
Done: https://github.com/GaloyMoney/galoy/pull/1535/commits/f44d455388e9d38ac8496e30e19c17dc78c04b65

Reply:
maybe I'm missing what is being implemented, but I had in mind having something else than `LedgerTransactionType.Payment`. 

maybe something like `LedgerTransactionType.PaymentReimbursement.`